### PR TITLE
Enforce KeyLock API constraints and synchronize READMEs

### PR DESCRIPTION
**Pull request by pmalirz**
This submission implements strict input validation for the `tryLock` method inside `SimpleKeyLock` to guarantee safe database operations and enforce documented API constraints. The changes reject empty/null/oversized lock keys and invalid expiration times, while properly handling `null` references during `unlock`. It includes unit test coverage for these edge cases. Additionally, documentation across the main `README.md`, `dlock-core/README.md`, and `dlock-api` interface Javadocs has been fully updated to explicitly outline these constraints, fulfilling the requirement for a clean, secure API and aligned documentation.

---
*PR created automatically by Jules for task [6709944941719295175](https://jules.google.com/task/6709944941719295175) started by @pmalirz*

### CODE REVIEW

File: dlock-core/src/main/java/io/github/pmalirz/dlock/core/SimpleKeyLock.java [hunk 1]
```diff
@@ -36,6 +36,13 @@ public SimpleKeyLock(LockRepository lockRepository,
 
     @Override
     public Optional<LockHandle> tryLock(String lockKey, long expirationSeconds) {
+        if (lockKey == null || lockKey.trim().isEmpty() || lockKey.length() > 1000) {
```
Comment (pmalirz) [inline dlock-core/src/main/java/io/github/pmalirz/dlock/core/SimpleKeyLock.java]: @jules - this 1000 magic number is pread across the codebase. Maybe would be good to define it one place.

Comment (pmalirz) [reply dlock-core/src/main/java/io/github/pmalirz/dlock/core/SimpleKeyLock.java]: I do not see the changes. This PR has only one commit

File: dlock-api/src/main/java/io/github/pmalirz/dlock/api/KeyLock.java [hunk 2]
```diff
@@ -11,16 +11,21 @@
  *
  * @author Przemyslaw Malirz
  */
-public interface KeyLock {
+public interface KeyLock {
+
+    int MAX_LOCK_KEY_LENGTH = 1000;
```
Comment (pmalirz) [inline dlock-api/src/main/java/io/github/pmalirz/dlock/api/KeyLock.java]: @jules - what with javadocs?

File: dlock-core/src/main/java/io/github/pmalirz/dlock/core/SimpleKeyLock.java (hunk 1 above)
Comment (pmalirz) [reply dlock-core/src/main/java/io/github/pmalirz/dlock/core/SimpleKeyLock.java]: @jules - 1000 is still a magic number in the javadocs

File: dlock-api/src/main/java/io/github/pmalirz/dlock/api/KeyLock.java (hunk 2 above)
Comment (pmalirz) [reply dlock-api/src/main/java/io/github/pmalirz/dlock/api/KeyLock.java]: this is bad, I can see in the current PR in javadocs plain 1000 instead of MAX_LOCK_KEY_LENGTH
